js/src/lib/util.js

```diff
@@ -110,3 +133,43 @@ function parseHighlightedAttribute(
 
   return elements;
 }
+
+export const packageLink = name =>
+  `${window.i18n.url_base}/package${process.env.NODE_ENV === 'production'
+    ? '/'
+    : '?'}${name}`;
+
+export const searchLink = query =>
+  `${window.i18n.url_base}/packages?q=${query}`;
+
+export const prefixURL = (url, { base, user, project, head, path }) => {
+  if (url.indexOf('//') > 0) {
+    return url;
+  } else {
+    return new URL(
+      (path ? path.replace(/^\//, '') + '/' : '') +
+        url.replace(/^(\.?\/?)/, ''),
+      `${base}/${user}/${project}/${path ? '' : `${head}/`}`,
+    );
+  }
+};
+
+const status = res => new Promise((resolve, reject) => {
+  if (res.status >= 200 && res.status < 300) {
+    if (res.status === 202 || res.status === 204) {
```

Comment:
>The GitHub API has another slight quirk, and that’s using HTTP status 202. This is used in the activity response when the data hasn’t been computed yet. Instead of making the request take a long time and maybe timeout, they choose to send that response with an empty JSON instead. This signals me that I should try again with a slight delay and see if the response has been computed in the meantime.

I wrote about this in my blog: https://haroen.me/algolia-blog/late